Create a documentation page for FX graph mode quantization APIs

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66222 Quantization docs: add pages for Numeric Suite (Eager and FX)
* #66210 Quantization documentation: move backend section down
* #66201 Quantization docs: rewrite API reference to be more automated
* #66198 Quantization docs: consilidate all API references on a single page
* #66129 Create a documentation page for `torch.ao.quantization.QConfig`
* #66125 Create separate documentation pages for quantization observers and fake_quants
* **#66122 Create a documentation page for FX graph mode quantization APIs**

Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

Differential Revision: [D31447612](https://our.internmc.facebook.com/intern/diff/D31447612)

vkuzo committed:
Create a documentation page for FX graph mode quantization APIs

Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for FX graph mode quantization APIs"


Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for FX graph mode quantization APIs"


Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for FX graph mode quantization APIs"


Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

[ghstack-poisoned]

jerryzh168 commented on docs/source/torch.quantization.quantize_fx.rst:
```diff
@@ -0,0 +1,24 @@
+.. _torch_quantization_quantize_fx:
```
should we rename this to torch.ao.quantization.quantize_fx.rst?

vkuzo replied:
I was thinking that we should avoid any communications about `torch.ao` in docs until there is an OSS facing post with the details, etc.  I.e once the migration phase 2 is complete, we would update all the docs after that.  But, I'm flexible.  cc @z-a-f .

Thankfully sphinx is able to link to the docblocks in `torch.ao`, even though we are exposing the old locations before the migration.

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31447612).

vkuzo committed:
Update on "Create a documentation page for FX graph mode quantization APIs"


Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

Differential Revision: [D31447612](https://our.internmc.facebook.com/intern/diff/D31447612)

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for FX graph mode quantization APIs"


Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

Differential Revision: [D31447612](https://our.internmc.facebook.com/intern/diff/D31447612)

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for FX graph mode quantization APIs"


Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

Differential Revision: [D31447612](https://our.internmc.facebook.com/intern/diff/D31447612)

[ghstack-poisoned]

vkuzo committed:
Update on "Create a documentation page for FX graph mode quantization APIs"


Description:

Adds a documentation page for FX graph mode quantization APIs which
reads from the docstrings in `quantize_fx`, and links it from the main
quantization documentation page.

Also, updates the docstrings in `quantize_fx` to render well with reStructuredText.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
```

Differential Revision: [D31447612](https://our.internmc.facebook.com/intern/diff/D31447612)

[ghstack-poisoned]
